types/socket.io/index.d.ts

```diff
@@ -81,7 +81,7 @@ interface SocketIOStatic {
 
 declare namespace SocketIO {
     interface Server {
-        engine: { ws: any };
+        engine: { ws: any, generateId: (id: string) => string };
```

Comment:
> The function is called with a node request object (http.IncomingMessage) as the first parameter.

Reply:
My apologies again I should read it better. I resolved it.